Sources/SentryCrash/Recording/SentryCrashCachedData.c

```diff
@@ -156,6 +171,15 @@ sentrycrashccd_init(int pollingIntervalInSeconds)
     pthread_attr_destroy(&attr);
 }
 
+void
+sentrycrashccd_close()
+{
+    if (g_hasThreadStarted == true) {
+        g_hasThreadStarted = false;
+        pthread_cancel(g_cacheThread);
+    }
+}
```

Comment:
`pthread_cancel` is a weird concept btw, it can do all kinds of funny things like leak memory, etc.
You should also either `pthread_join` to block end ensure cancelation, or `pthread_detach` the thread for a "fire and forget".

Not quite sure what this thread does, and if you can explicitly signal it to shut itself down? Usually threads should terminate from within, to make sure they free up all the resources they are using. As you say this is only used for testing and never in production, it might not matter that much though.

Reply:
@Swatinem, the thread runs an infinite loop, see https://github.com/getsentry/sentry-cocoa/blob/d68e2b6e0e62b0e26fede487eb5c2642825fa3fe/Sources/SentryCrash/Recording/SentryCrashCachedData.c#L125-L141

I'm afraid `pthread_detach` would not work. We have to terminate the thread somehow.

Reply:
Oh nice, it is detached already: https://github.com/getsentry/sentry-cocoa/blob/d68e2b6e0e62b0e26fede487eb5c2642825fa3fe/Sources/SentryCrash/Recording/SentryCrashCachedData.c#L150

Also, since you have `sleep` in there, that is basically defined as a cancellation point. So you should be good in that case. As long as the thread is not holding on to things across cancellation points; in which case you would leak. But looking at the thread itself, things should be good.

Reply:
Just to be sure, you are basically saying the code is fine as it is, @Swatinem?

Reply:
Yes, I think it is